scripts/zkai_proof_pressure_scaling_claim_pack_gate.py

```
@@ -0,0 +1,983 @@
+            results.append({"name": name, "rejected": False, "error": ""})
+    return results
+
+
+def validate_source_artifacts(payload: dict[str, Any]) -> None:
+    artifacts = require_list(payload.get("source_artifacts"), "source artifacts")
+    ids = set()
+    for index, artifact_any in enumerate(artifacts):
+        artifact = require_dict(artifact_any, f"source_artifacts[{index}]")
+        artifact_id = string_field(artifact.get("id"), f"source_artifacts[{index}].id")
+        if artifact_id in ids:
+            raise ProofPressureScalingClaimPackError("duplicate source artifact id")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)